cremona/src/Book.ts

```diff
@@ -1,43 +1,58 @@
 export class Book {
-    readonly name: string;
-    readonly uid: number;
-    readonly articleNumber: string;
-    readonly description: string | null;
-    readonly price: number;
-    readonly url: string[];
-    readonly image: string;
-    readonly courseCodes: string[];
-    readonly authors: string[];
-    readonly isbn: number | null;
-    readonly weight: number;
-    readonly year: number | null;
+  readonly name: string;
+  readonly uid: number;
+  readonly articleNumber: string;
+  readonly description: string | null;
+  readonly price: number;
+  readonly url: string[];
+  readonly image: string;
+  readonly courseCodes: string[];
+  readonly authors: string[];
+  readonly isbn: number | null;
+  readonly weight: number;
+  readonly year: number | null;
+  readonly publisher: string | null;
+  readonly edition: number | null;
 
-    constructor(props: { [key: string]: any }) {
-        this.name = props.name.sv;
-        this.uid = props.uid;
-        this.articleNumber = props.articleNumber;
-        try {
-            this.description = props.description.sv.replace(/(<([^>]+)>)/gi, ' ') // strip out HTML tags
-        } catch (error) {
-            this.description = null;
-        }
-        this.price = props.price.current.SEK;
-        this.url = props.url;
-        this.image = props.images[0];
-        props['attributes'][1424] !== undefined ? // convert string of course codes to list
-            this.courseCodes = props['attributes'][1424]['sv'].replace(/ /g, '').split(',') :
-            this.courseCodes = [];
-        props['attributes'][1357] !== undefined ?
-            this.authors = props['attributes'][1357]['sv'].replace(', ', ',').split(',') :
-            this.authors = [];
-        props['attributes'][1422] !== undefined ?
-            this.year = parseInt(props['attributes'][1422]['sv']) :
-            this.year = null;
-        if (props.ean)
-            this.isbn = parseInt(props.ean);
-        else
-            this.isbn = null;
-        this.weight = props.weight;
+  constructor(props: { [key: string]: any }) {
+    this.name = props.name.sv;
+    this.uid = props.uid;
+    this.articleNumber = props.articleNumber;
+    try {
+      this.description = props.description.sv.replace(/(<([^>]+)>)/gi, ' '); // strip out HTML tags
+    } catch (error) {
+      this.description = null;
     }
+    this.price = props.price.current.SEK;
+    this.url = props.url;
+    this.image = props.images[0];
+    if (props['attributes'][1357] !== undefined) {
+      this.authors = props['attributes'][1357]['sv']
+        .replace(', ', ',')
+        .split(',');
+    } else if (props['attributes'][2513] !== undefined) {
+      this.authors = props['attributes'][2513]['sv']
+        .replace(', ', ',')
+        .split(',');
+    } else {
+      this.authors = [];
```

Comment:
Shouldn't authors = null in order to stay consistent with all the empty attributes.

Reply:
Not sure here: @EliasFalk?

Reply:
I think I agree with @Danryg on this one. Otherwise strings should probably be empty strings and numbers be something like 0 in order to stay consistent and I think that's going to be more problematic. Not sure whether it would be possible or beneficial in this case but making the attributes for the class/interface for a book optional, for example `{authors?: Array}`, could also be a solution since it would either be an array or "undefined" and thus staying consistent that way.